client/my-sites/people/team-list/team.jsx

```diff
@@ -0,0 +1,136 @@
+import PeopleListItem from 'my-sites/people/people-list-item';
+import UsersActions from 'lib/users/actions';
+import InfiniteList from 'components/infinite-list';
+import deterministicStringify from 'lib/deterministic-stringify';
+import NoResults from 'my-sites/no-results';
+import PeopleListSectionHeader from 'my-sites/people/people-list-section-header';
+import { recordGoogleEvent } from 'state/analytics/actions';
+
+/**
+ * Module Variables
+ */
+export class Team extends Component {
+	static propTypes = {
+		translate: PropTypes.func,
+	};
+
+	static defaultProps = {
+		translate: identity,
+	};
+
+	isLastPage = () => this.props.totalUsers <= this.props.users.length + this.props.excludedUsers.length;
+
+	renderPerson = ( user ) => {
+		return (
+			<PeopleListItem
+				key={ user.ID }
+				user={ user }
+				type="user"
+				site={ this.props.site } />
+		);
+	};
+
+	fetchNextPage = () => {
+		const offset = this.props.users.length;
+		const fetchOptions = { ...this.props.fetchOptions, offset };
+		this.props.recordFetchUsersForOffset( offset );
+		debug( 'fetching next batch of users' );
+		UsersActions.fetchUsers( fetchOptions );
+	};
+
+	getPersonRef = ( user ) => `user-${ user.ID }`;
+
+	headerText = () => {
+		const { props } = this;
+		const { translate } = props;
+		if ( props.search && props.totalUsers && props.site && props.users.length ) {
+			return translate(
+				'%(numberPeople)d Person Matching {{em}}"%(searchTerm)s"{{/em}}',
+				'%(numberPeople)d People Matching {{em}}"%(searchTerm)s"{{/em}}',
+				{
+					count: props.users.length,
+					args: {
+						numberPeople: props.totalUsers,
+						searchTerm: props.search
+					},
+					components: {
+						em: <em />
+					}
+				}
+			);
+		}
+		return translate( 'Team' );
+	};
+
+	render() {
+		const { props } = this;
+		const { translate, users, fetchInitialized, fetchOptions, fetchingUsers } = props;
+
+		if ( fetchInitialized && ! users.length && fetchOptions.search && ! fetchingUsers ) {
+			return (
+				<NoResults
+					image="/calypso/images/people/mystery-person.svg"
+					text={
+						translate( 'No results found for {{em}}%(searchTerm)s{{/em}}',
+							{
+								args: { searchTerm: props.search },
+								components: { em: <em /> }
+							}
+						)
+					} />
+			);
+		}
+
+		return (
+			<div>
+				<PeopleListSectionHeader
+					label={ this.headerText() }
+					site={ props.site }
+					count={ props.fetchingUsers || props.fetchOptions.search ? null : props.totalUsers } />
+				<Card>
+					<InfiniteList
+						key={ deterministicStringify( omit( fetchOptions, [ 'number', 'offset' ] ) ) }
+						items={ users }
+						className="people-selector__infinite-list"
+						ref="infiniteList"
+						fetchingNextPage={ fetchingUsers }
+						lastPage={ this.isLastPage() }
+						fetchNextPage={ this.fetchNextPage }
+						getItemRef={ this.getPersonRef }
+						renderLoadingPlaceholders={ () => <PeopleListItem key="people-list-item-placeholder" /> }
```

Comment:
there should have been a linting warning about this - creating closures from inside JSX parameters. this will create a new closure on every call to `render()`. it would be much better, especially since this function returns a static value, to simply define it elsewhere, ideally in the module scope outside of the React element.